### PR TITLE
External storage post merge work

### PR DESCRIPTION
**Pull request by wmathurin**
Code to prevent soup from having json1 index and external storage (and new test for it)
Test code duplication cleanup
New tests for externally stored soup with full text
New tests for alter soup with storage change and column type changes 

NB: The hybrid/react native work will come in a separate PR

### CODE REVIEW

**Comment by sgoldberg-sfdc** [comment]
👍  Thanks for cleaning that up.

File: libs/SmartStore/SmartStoreTests/SFSmartStoreFullTextSearchTests.h [hunk 1]
```diff
@@ -27,5 +27,12 @@
 
 #import "SFSmartStoreTestCase.h"
 
-@interface SFSmartStoreFullTextSearchTests : SFSmartStoreTestCase 
+@interface SFSmartStoreFullTextSearchTests : SFSmartStoreTestCase
+
+#define kTestStore            @"testSmartStoreFullTextSearchStore"
```
Comment (bhariharan) [inline libs/SmartStore/SmartStoreTests/SFSmartStoreFullTextSearchTests.h]: I wonder if all the ```#define``` should be ```static const NSString *``` instead. Minor difference, but I think we do the latter in our code usually. I don't care much either way.

**Comment by bhariharan** [comment]
Looks pretty good otherwise.

File: libs/SmartStore/SmartStoreTests/SFSmartStoreFullTextSearchTests.h (hunk 1 above)
Comment (wmathurin) [inline libs/SmartStore/SmartStoreTests/SFSmartStoreFullTextSearchTests.h]: I have only be using #define in tests (and usually in the .m). I agree we should use const in the code. For tests, I don't think it matters.

Comment (bhariharan) [inline libs/SmartStore/SmartStoreTests/SFSmartStoreFullTextSearchTests.h]: 👍

Comment (kchitalia) [inline libs/SmartStore/SmartStoreTests/SFSmartStoreFullTextSearchTests.h]: Just for my understanding, what's the difference between  #define ans static const NSString * ?

Comment (bhariharan) [inline libs/SmartStore/SmartStoreTests/SFSmartStoreFullTextSearchTests.h]: @kchitalia ```#define``` is a preprocessor directive, so basically it's a macro - fancy way of doing ```Find and Replace```. Every occurrence of that key is substituted by the value. ```static const``` creates a single copy of that variable, and allocates memory to it, which is common to all instances of that object, so it's always in memory. That's basically the difference.

File: libs/SmartStore/SmartStoreTests/SFSmartSqlWithExternalStorageTests.m [hunk 2]
```diff
@@ -30,6 +30,29 @@
 #import <SalesforceSDKCore/SFJsonUtils.h>
 #import "SFSoupSpec.h"
 
+@interface SFSmartSqlTests ()
```
Comment (khawkins) [inline libs/SmartStore/SmartStoreTests/SFSmartSqlWithExternalStorageTests.m]: Not the end of the world, but in a scenario like this, why not put responsibility of defining the interface on the parent class, then reference that interface here?

Comment (wmathurin) [inline libs/SmartStore/SmartStoreTests/SFSmartSqlWithExternalStorageTests.m]: That makes sense, I just made the change here: https://github.com/forcedotcom/SalesforceMobileSDK-iOS/pull/1650/commits/d8b65aacd8f24d6e52c4b058d24829de22283783

**Pull request merged**